### DISCUSSION TITLE
Connecting to Planetscale on Windows via Python Cert Issues

### DISCUSSION THREAD

**Discussion by deshumake** [planetscale/discussion#431] [Q&A]
I was reading PlanetScale's article for how to connect to a database using python, and am not understanding what's wrong with my certificate. Could someone help, or is there other documentation I'm missing?

This is the article I'm following: https://planetscale.com/blog/connect-to-a-mysql-database-in-python
I am currently using PyMySQL, but none of the options have worked for me yet. This is the example code they've given
```
from dotenv import load_dotenv
import pymysql
import os
load_dotenv()
connection = pymysql.connect(
    host=os.getenv("HOST"),
    database=os.getenv("DATABASE"),
    user=os.getenv("USERNAME"),
    password=os.getenv("PASSWORD"),
    ssl_ca=os.getenv("SSL_CERT")
)
```

I have an env file that looks like this:
```
HOST=aws.connect.psdb.cloud
USERNAME=###
PASSWORD=###
DATABASE=mydb
SSL_CERT="C:\path\cacert.pem"
```

I downloaded the cacert.pem file from https://curl.se/docs/caextract.html, as was instructed in this article https://planetscale.com/docs/concepts/secure-connections#ca-root-configuration.

Am I referencing the path incorrectly? I keep getting an error that says access is denied. I assume it is because of the certificate because all other information was copied from PlanetScale when I clicked the connect button. Is there anything else I may be missing?

**Comment by fooforge** [comment]
@deshumake Hey 👋,

This is most likely due to your `SSL_CERT` environment variable. Python's parser reserves `\` for escape sequences so for an actual `\` to appear in your string, you will need to escape it first. Like so:

```python
SSL_CERT="C:\\path\\cacert.pem"
```

See [Python's documentation on lexical analysis](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals) for more information on escape sequences.

**Reply by fooforge** [reply]
@deshumake Thinking about this some more, `os.getenv` handles the escaping already so it's likely something else. Mind sharing the exact error message that you're getting?

```python
➜  ~ export TMP_SSL_CERT="C:\path-to-ssl-cert\cacert.pem"
➜  ~ python3
Python 3.9.6 (default, Mar 10 2023, 20:16:38)
[Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> value = os.getenv("TMP_SSL_CERT")
>>> print(value)
C:\path-to-ssl-cert\cacert.pem
>>> print(repr(value))
'C:\\path-to-ssl-cert\\cacert.pem'
```
